### PR TITLE
add redact option for env block when pushing an app with a manifest.yml

### PR DESCRIPTION
**Pull request by nouseforaname**
IF the app is pushed with a manifest
    AND the app needs secrets set via environment variables
    THEN currently there is no way to avoid leaking secrets in the output of
    cf push commands.

    technically this could be avoided by not using a manifest and running:
    ```cf push --no-start
    cf set-env ...
    cf set-env ...
    cf set-env ...
    ...
    `

    but that escalates quickly if the APP contains many secret things in it's
    env requirements.

    We face this in an errand where:
    - the bosh release renders the application.yml
    - the application.yml contains secrets in the env block
    - the errand VM streams it's logs to a logging server

    This can be remediated by the suggested redacttion.
    `--redact-env` to indiscriminately change all values
    to `<redacted>` if they're contained within an apps `env` block.

### CODE REVIEW

**Review by a-b — APPROVED** [review]
Thanks for your contribution. LGTM!

**Comment by a-b** [comment]
@nouseforaname, If you want to see this change in the next V8 release, please create a pull request for the respective branch.

File: integration/v7/push/manifest_diff_test.go [hunk 1]
```diff
@@ -53,4 +53,43 @@ var _ = Describe("displaying manifest differences between pushes", func() {
 			})
 		})
 	})
+	FWhen("there is an app that has been pushed and we push it again with a different manifest", func() {
```
Comment (a-b) [inline integration/v7/push/manifest_diff_test.go:56]: @nouseforaname, can you please remove the focus?

File: integration/v7/push/manifest_diff_test.go [hunk 2]
```diff
@@ -53,4 +53,43 @@ var _ = Describe("displaying manifest differences between pushes", func() {
...
+		FIt("displays the diff from the manifest", func() {
```
Comment (a-b) [inline integration/v7/push/manifest_diff_test.go:57]: Same focus

**Review by a-b — CHANGES_REQUESTED** [review]
Please remove the focus on tests.

Also we should ensure the linter catches issues earlier.

cc @moleske

**Comment by nouseforaname** [comment]
> @nouseforaname, If you want to see this change in the next V8 release, please create a pull request for the respective branch.

That is actually an interesting point. Should this go anywhere else than v8?

File: integration/v7/push/manifest_diff_test.go (hunk 1 above)
Comment (nouseforaname) [reply integration/v7/push/manifest_diff_test.go]: Done, thanks for spotting this.

File: integration/v7/push/manifest_diff_test.go (hunk 2 above)
Comment (nouseforaname) [reply integration/v7/push/manifest_diff_test.go]: Done, thanks for spotting this.

**Comment by moleske** [comment]
> That is actually an interesting point. Should this go anywhere else than v8?

v7 is the only other place for this to potentially go.  So if needed on the v7 cli than a pr to the v7 branch would be needed

**Review by a-b — APPROVED** [review]
No more focused tests. LGTM!

**Pull request merged**